### PR TITLE
Run the benchmarks in the unit test

### PR DESCRIPTION
**Pull request by JaroslavTulach**
I'd like to make sure that the unit tests are executed at "Graal-speed". First step is to measure how fast the engines are. Next step is to configure the plugin to use the Graal compiler.

### CODE REVIEW

**Comment by boris-spas** [comment]
Good luck! I've tried to make that happen but have unfortunately failed.

**Comment by JaroslavTulach** [comment]
It needed a lot of debugging of the Maven surefire plugin. There is special handling for `module-info` in main source directory. Alas, I don't see special handling of `module-info` in test sources. As such I had to add `truffle-api` into `compile` category. That isn't ideal, but it works.

**Comment by chumer** [comment]
Its not ideal that we need to put it on compile. Is that a bug of Maven surefire that it does not respect transitive dependencies? Because js refers to truffle already?

**Comment by JaroslavTulach** [comment]
So far I managed to convince surefire to run in "module mode" for the main project sources and "classpath mode" for the tests. In this setup the `truffle-api` module has to be shared between the compiler and the JavaScript engine. Using `compile` scope does the trick. So far other attempts failed.

**Comment by JaroslavTulach** [comment]
At the end it was not that complex. No need to play with `module-info.java` at all. Just download SDK, Truffle API and compiler into a subdirectory and enable them as modules.

**Comment by boris-spas** [comment]
Very cool! Thanks a lot!

File: pom.xml [hunk 1]
```diff
@@ -125,6 +139,12 @@
             <artifactId>graal-sdk</artifactId>
             <version>${graalvm.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.graalvm.truffle</groupId>
+            <artifactId>truffle-api</artifactId>
+            <version>${graalvm.version}</version>
+            <scope>runtime</scope>
```
Comment (JaroslavTulach) [inline pom.xml]: Runtime only dependency. Possibly redundant, as `org.graalvm.js:js` would probably bring this one in anyway.

Comment (JaroslavTulach) [reply pom.xml]: But is it true this `pom.xml` could be shorter: proposed in [PR-7](https://github.com/graalvm/graal-js-jdk11-maven-demo/pull/7/).

**Pull request merged**